### PR TITLE
Generalize and update script

### PR DESCRIPTION
**Pull request by whoabuddy**
This PR aims to generalize the script so that anyone can use it by modifying the variables at the top, as well as follow some bash3 boilerplate standards.

This will also add a 1hr loop for publishing on a regular basis using the logged in user's git credentials.

### CODE REVIEW

File: publish.sh [hunk 1]
```diff
@@ -1,11 +1,80 @@
...
+if [ "$__action" == "" ]; then 
+  publish
+  exit
+else
+  while true
+  do
+    __timestamp=$(date -u)
+    publish
+    printf "Published at: %s", "$__timestamp"
+  done
```
Comment (friedger) [inline publish.sh:64]: Shouldn't the script sleep for an hour here?

Comment (whoabuddy) [reply publish.sh:64]: Yep! See f078880, I know I forget to sleep sometimes too :zzz:

**Comment by whoabuddy** [comment]
I think I hit all the points but haven't tested this out yet, so you may want to give it a go before merging. It's set up to use your defaults still, but leaves some room for others to customize based on their own configuration.

**Comment by friedger** [comment]
Thank you!

**Pull request merged**